std/range/primitives.d

```diff
@@ -2104,28 +2104,28 @@ if (isNarrowString!(C[]))
 
     static if (is(Unqual!C == char))
     {
+        __gshared static immutable ubyte[] charWidthTab = [
+            2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+            2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+            3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
+            4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 1, 1
+        ];
+        /*
+          This implementation of popFront does not special case 0b10xx_xxxx
+          Nither did the one before.
+          on illigal strings we may skip too many bytes.
+         */
```

Comment:
I don't understand this comment. So if the first byte has the shape `0x10xx_xxxx` (illegal) we skip one byte which is what we need to do.